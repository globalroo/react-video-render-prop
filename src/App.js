import React, { Component } from "react";
import logo from "./logo.svg";
import Video from "./Video";
import "./App.css";

class App extends Component {
  test() {
    return "World";
  }
  render() {
    const hi = "Hello";
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Render prop video player</h1>
        </header>
        <p className="App-intro">
          Testing render props with a video component.
        </p>
        <Video
          renderProp={state => (
            <div>
              <h3>
                Now have access to parent context {hi} {this.test()}
              </h3>
              <div> Current Time: {state.currentTime} </div>
            </div>
          )}
        />
      </div>
    );
  }
}

export default App;
