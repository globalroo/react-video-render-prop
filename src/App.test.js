import React from "react";
import App from "./App";
import * as renderer from "react-test-renderer";

function handleDOMMocks(element) {
  if (element.type === "video") {
    return {
      addEventListener: jest.fn()
    };
  }
  return null;
}
describe("App test", () => {
  test("App renders as expected", () => {
    const options = {
      createNodeMock: handleDOMMocks
    };
    const component = renderer.create(<App />, options);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
