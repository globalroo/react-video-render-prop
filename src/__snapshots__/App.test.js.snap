// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test App renders as expected 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Render prop video player
    </h1>
  </header>
  <p
    className="App-intro"
  >
    Testing render props with a video component.
  </p>
  <div>
    <video
      controls={true}
    />
    <div>
      <h3>
        Now have access to parent context 
        Hello
         
        World
      </h3>
      <div>
         Current Time: 
        0
         
      </div>
    </div>
  </div>
</div>
`;
